inginious/frontend/pages/course_admin/student_list.py

```diff
@@ -284,6 +292,37 @@ def post_groups(self, course, data, active_tab, msg, error):
                 msg["groups"] = _('An error occurred while parsing the data.')
                 error["groups"] = True
             active_tab = "tab_groups"
+        if "audiencecreationfile" in data and 'upload_audiences_creation' in data:
+            reader = csv.reader(data["audiencecreationfile"].read().decode("utf-8").split("\n"))
+            for line in reader:
+                if len(line)!=4:
+                    msg["audiences"] = _("File wrongly formatted.")
+                    error["audiences"] = True
+            if not error["udiences"]:
+                stud_list, _, _, _ = self.get_user_lists(course)
+                self.database.audiences.delete_many({"courseid": course.get_id()})
+                for user_id, field, role, description in reader:
+                    user_id = user_id.strip()
+                    field = field.strip()
+                    role = role.strip()
+                    description = description.strip()
+                    query = {"courseid": course.get_id(), "description": description}
+                    if field is not "username":
```

Comment:
``field`` should be validated before performing the request. Otherwise, any field can be matched, and Mongo operators can be used.